lib/openstudio-standards/standards/ashrae_90_1/ashrae_90_1_2007/179d_ashrae_90_1_2007/179d_ashrae_90_1_2007.ZoneHVACComponent.rb

```diff
@@ -0,0 +1,139 @@
+      zone_hvac_component = zone_hvac_component.to_ZoneHVACTerminalUnitVariableRefrigerantFlow.get
+      if zone_hvac_component.outdoorAirFlowRateWhenNoCoolingorHeatingisNeeded.is_initialized
+        oa_rate = zone_hvac_component.outdoorAirFlowRateWhenNoCoolingorHeatingisNeeded.get
+        ventilation = true if oa_rate > 0.0
+      end
+      ventilation = true if zone_hvac_component.isOutdoorAirFlowRateWhenNoCoolingorHeatingisNeededAutosized
+      existing_sch = zone_hvac_component.supplyAirFanOperatingModeSchedule
+    elsif zone_hvac_component.to_ZoneHVACWaterToAirHeatPump.is_initialized
+      zone_hvac_component = zone_hvac_component.to_ZoneHVACWaterToAirHeatPump.get
+      if zone_hvac_component.outdoorAirFlowRateWhenNoCoolingorHeatingisNeeded.is_initialized
+        oa_rate = zone_hvac_component.outdoorAirFlowRateWhenNoCoolingorHeatingisNeeded.get
+        ventilation = true if oa_rate > 0.0
+      end
+      ventilation = true if zone_hvac_component.isOutdoorAirFlowRateWhenNoCoolingorHeatingisNeededAutosized
+      fan_op_sch = zone_hvac_component.supplyAirFanOperatingModeSchedule
+      existing_sch = fan_op_sch.get if fan_op_sch.is_initialized
+    end
+    return false unless ventilation
+
+    # if supply air fan operating schedule is always off,
+    # override to provide ventilation during occupied hours
+    # NOTE: 179D disable due to override with acm schedule
+    # unless existing_sch.nil?
+    #   if existing_sch.name.is_initialized
+    #     OpenStudio.logFree(OpenStudio::Info, 'openstudio.Standards.ZoneHVACComponent', "#{zone_hvac_component.name} has ventilation, and schedule is set to always on; keeping always on schedule.")
+    #     return false if existing_sch.name.get.to_s.downcase.include?('always on discrete') || existing_sch.name.get.to_s.downcase.include?('guestroom_vent_ctrl_sch')
+    #   end
+    # end
+
+    # thermal_zone = zone_hvac_component.thermalZone.get
+    # occ_threshold = zone_hvac_unoccupied_threshold
+    # occ_sch = thermal_zones_get_occupancy_schedule([thermal_zone],
+    #                                                sch_name: "#{zone_hvac_component.name} Occ Sch",
+    #                                                occupied_percentage_threshold: occ_threshold)
+    ## NOTE: using 179d acm schedule
+    data = model_get_standards_data(zone_hvac_component.model, throw_if_not_found: true)
+    acm_fan_sch_name = data['hvac_operation_schedule']
+    zone_hvac_component.model.getBuilding.additionalProperties.setFeature('acm_fan_sch', acm_fan_sch_name)
+    acm_fan_sch = model_add_schedule(zone_hvac_component.model, acm_fan_sch_name)
+    zone_hvac_component.setSupplyAirFanOperatingModeSchedule(acm_fan_sch)
+    OpenStudio.logFree(OpenStudio::Info, 'openstudio.Standards.ZoneHVACComponent', "#{zone_hvac_component.name} has ventilation.  Setting fan operating mode schedule to ACM schedule '#{acm_fan_sch_name}'.")
```

Comment:
Now I use the model_get_standards_data to grab the hvac_operation_schedule name from the hash.